test/common/policy_utils.go

```diff
@@ -166,29 +168,36 @@ func DoRootComplianceTest(policyName string, compliance policiesv1.ComplianceSta
 	).Should(Equal(compliance))
 }
 
-// GetLatestStatusMessage returns the most recent status message for the given policy template.
-// If the policy, template, or status do not exist for any reason, an empty string is returned.
-func GetLatestStatusMessage(policyName string, templateIdx int) func() string {
-	return func() string {
-		replicatedPolicyName := UserNamespace + "." + policyName
-		policyInterface := ClientManagedDynamic.Resource(GvrPolicy).Namespace(ClusterNamespace)
+func GetHistoryMessages(policyName string, templateIdx int) ([]interface{}, bool, error) {
+	empty := make([]interface{}, 0)
+	replicatedPolicyName := UserNamespace + "." + policyName
+	policyInterface := ClientManagedDynamic.Resource(GvrPolicy).Namespace(ClusterNamespace)
 
-		policy, err := policyInterface.Get(context.TODO(), replicatedPolicyName, metav1.GetOptions{})
-		if err != nil {
-			return ""
-		}
+	policy, err := policyInterface.Get(context.TODO(), replicatedPolicyName, metav1.GetOptions{})
+	if err != nil {
+		return empty, false, errors.New("error in getting policy")
```

Comment:
Nitpick: instead of `errors.New`, this could wrap the error like `fmt.Errorf("failed to get policy for history: %w", err)`. This would give us a little more information if it fails during a test.